### PR TITLE
Resolved issue #2

### LINKED ISSUE #2: Add Delete feature to Ticket Card
Add a feature to delete the card whenever user want so that it will not get visible on dashboard and gets permanently deleted form database

### PR DESCRIPTION
**Pull request by MadhavDhall**
### Related Issue  
Closes #2 

### Type of Change
- [x] New Feature  
- [ ] Bug Fix  
- [ ] Code Refactor  
- [ ] Documentation Update  
- [ ] Other (please specify): 

### Description of Change  
A delete button is added in the update ticket page to delete that ticket. Before deleting the ticket, it requires a confirmation to delete the ticket.

### Implementation Details  
I have added a button to delete the ticket which is visible only when the ticket is getting updated and not visible when creating a new ticket. When the **Delete** button is clicked, a confirmation box pops up. On confirming, a fetch request is sent to **`/api/Tickets/${ticket._id}`**. The ticket gets permanently deleted from the database and the user is returned to the homepage with router.push("/") where the deleted ticket no longer exists.

### Demo  

https://github.com/user-attachments/assets/4d876ee7-2d54-485a-bb24-e64a9f9087df

### CODE REVIEW

**Comment by AyushDharDubey** [comment]
Hi @MadhavDhall, congratulations on your first contribution to Ticketify, your implementation looks good, and we appreciate your effort.

one important thing to keep in mind for future contributions: contributors should always discuss their proposed implementation on the issue page and wait for clearance from the maintainers before opening a new PR to resolve the issue. This ensures alignment and smooth collaboration.

i am suggesting some minor changes below, please incorporate them and then your PR will be ready to be merged.

File: app/(components)/EditTicketForm.jsx [hunk 1]
```diff
@@ -194,11 +221,18 @@ const EditTicketForm = ({ ticket }) => {
...
-          className="btn max-w-xs"
-          value={EDITMODE ? "Update Ticket" : "Create Ticket"}
-        />
+        <div className="flex flex-row flex-wrap justify-between gap-2">
+          {/* <input
+            type="submit"
+            className="btn max-w-xs"
+            value={EDITMODE ? "Update Ticket" : "Create Ticket"}
+          /> */}
+          <button type="submit" className="btn max-w-xs">
```
Comment (AyushDharDubey) [inline app/(components)/EditTicketForm.jsx:230]: this is an unnecessary change, the commented-out section of code:

```javascript
          {/* <input
            type="submit"
            className="btn max-w-xs"
            value={EDITMODE ? "Update Ticket" : "Create Ticket"}
          /> */}
```

is already functional and serves its purpose. Therefore, this change is unnecessary. Please refrain from committing changes that do not contribute to resolving the issue at hand.

Once this is addressed, your PR will be ready for merging.
Thank you

**Review by AyushDharDubey — CHANGES_REQUESTED** [review]
additionally all 3 lines changed in `app/page.jsx` can be categorised as refactoring changes and are unrelated to the concerned issue
so this file should be restored to the original

File: app/(components)/EditTicketForm.jsx (hunk 1 above)
Comment (MadhavDhall) [reply app/(components)/EditTicketForm.jsx:230]: Sir actually the thing was that input and button had different styles and their heights were different, hence it was looking odd and not good. 
Therefore I converted input into a button.
Rest as you say 😊

**Review by MadhavDhall — COMMENTED** [review]
Done changes as said by reviewer.

**Comment by AyushDharDubey** [comment]
Thank you @MadhavDhall, this PR will be merged soon, stay tuned
